include/pybind11/eigen.h

```diff
@@ -346,7 +346,7 @@ struct type_caster<Type, enable_if_t<is_eigen_dense_plain<Type>::value>> {
 
     operator Type*() { return &value; }
     operator Type&() { return value; }
-    operator Type&&() && { return std::move(value); }
+    operator Type() { return std::move(value); }
```

Comment:
Do we want to replace `operator Type&&` with `operator Type` or have both? `PYBIND11_TYPE_CASTER` defines both.

Reply:
You are right. We should keep both, one for move-forwarding and one for move construction.